src/main/java/io/pravega/connectors/flink/table/catalog/pravega/PravegaCatalog.java

```diff
@@ -72,41 +70,30 @@
     private SchemaRegistryClient schemaRegistryClient;
+    private ClientConfig clientConfig;
     private SchemaRegistryClientConfig config;
-    private final URI controllerUri;
-    private final URI schemaRegistryUri;
     private Map<String, String> properties;
     private SerializationFormat serializationFormat;
 
-    public PravegaCatalog(String catalogName, String defaultDatabase, String controllerUri, String schemaRegistryUri,
-                          String serializationFormat, String failOnMissingField, String ignoreParseErrors,
-                          String timestampFormat, String mapNullKeyMode,
-                          String mapNullKeyLiteral, String encodeDecimalAsPlainNumber) {
+    /**
+     * Creates a new Pravega Catalog instance.
+     *
+     * @param catalogName             The Pravega catalog name.
+     * @param defaultDatabase         The default database for Pravega catalog, which is mapped to Pravega scope here.
+     * @param properties              The options read from configuration.
+     * @param clientConfig            The Pravega client configuration.
+     * @param config                  The Schema Registry client configuration.
+     * @param serializationFormat     The serialization format used for serialization.
+     */
+    public PravegaCatalog(String catalogName, String defaultDatabase, Map<String, String> properties,
+                          ClientConfig clientConfig, SchemaRegistryClientConfig config,
+                          SerializationFormat serializationFormat) {
 
         super(catalogName, defaultDatabase);
-
-        this.controllerUri = URI.create(controllerUri);
-        this.schemaRegistryUri = URI.create(schemaRegistryUri);
-        this.config = SchemaRegistryClientConfig.builder()
-                .schemaRegistryUri(this.schemaRegistryUri).build();
-        this.serializationFormat = SerializationFormat.valueOf(serializationFormat);
-
-        this.properties = new HashMap<>();
-        properties.put(FactoryUtil.CONNECTOR.key(), FlinkPravegaDynamicTableFactory.IDENTIFIER);
-        properties.put(PravegaOptions.CONTROLLER_URI.key(), controllerUri);
-        properties.put(FactoryUtil.FORMAT.key(), PravegaRegistryFormatFactory.IDENTIFIER);
-        properties.put(
-                String.format(
-                        "%s.%s",
-                        PravegaRegistryFormatFactory.IDENTIFIER, PravegaRegistryOptions.URI.key()),
-                schemaRegistryUri);
-
-        properties.put(String.format("%s.%s",
-                PravegaRegistryFormatFactory.IDENTIFIER, PravegaRegistryOptions.FORMAT.key()),
-                this.serializationFormat.name());
-
-        propagateJsonOptions(failOnMissingField, ignoreParseErrors, timestampFormat,
-                mapNullKeyMode, mapNullKeyLiteral, encodeDecimalAsPlainNumber);
+        this.clientConfig = clientConfig;
+        this.config = config;
+        this.serializationFormat = serializationFormat;
```

Comment:
I prefer the previous way, pass the string there and change it to `SerializationFormat` here.